reyna/src/com/b2msolutions/reyna/Dispatcher.java

```diff
@@ -52,17 +61,32 @@ protected Result sendMessage(Message message, HttpPost httpPost, HttpClient http
     public static Result canSend(Context context) {
         ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
         NetworkInfo info = connectivityManager.getActiveNetworkInfo();
-        if(info == null || !info.isConnectedOrConnecting()) {
+        if (info == null || !info.isConnectedOrConnecting()) {
             return Result.NOTCONNECTED;
         }
+        int type = info.getType();
```

Comment:
We refactored again following @aflugge  suggestion, we were at the current point after a long refactoring so the code got more complicated than it needs to be.
Please @Youhana-Hana have a look now.

@lazywithclass